src/client/wire_protocol/flags.rs

```diff
@@ -0,0 +1,106 @@
+pub struct OpReplyFlags {
+    cursor_not_found: bool, // Bit 0
+    query_failure: bool,    // Bit 1
+    await_capable: bool,    // Bit 3
+
+    // All bits remaining must be 0
+}
+
+impl OpReplyFlags {
+    pub fn from_i32(i: i32) -> OpReplyFlags {
+        let cnf = (i & 1) != 0;
```

Comment:
General naming convention: don't use acronyms unless they're well-known, like 'http' or 'bson'. Stick to 'cursor_not_found' as the var name